### ISSUE TITLE
Implement Query Classification and Data Catalog Generation Using Anthropic Classification Methodology

### ISSUE THREAD

**Issue by stharrold** [stharrold/yuimedi-paper-20250901#12]
# Implement Query Classification and Data Catalog Generation Using Anthropic Classification Methodology

## Objective
Document YuiQuery's classification capabilities for query routing and data catalog automation, referencing Anthropic's classification guide as technical methodology baseline.

## Source Document
- **URL**: https://github.com/anthropics/anthropic-cookbook/blob/main/skills/classification/guide.ipynb
- **Type**: Technical implementation guide (Jupyter notebook)
- **Publisher**: Anthropic
- **Access Date**: [Current Date]

### 1. Query Classification by Business Context
**Purpose**: Route incoming queries to appropriate processing pipelines

#### Classification Taxonomies [Require Internal Definition]
- **Business Initiative Categories**:
  - Quality improvement programs [Define specific programs]
  - Value-based care metrics [List actual metrics]
  - Regulatory compliance reporting [Specify regulations]
  - Population health management [Define scope]
  - [Internal Doc: Business Taxonomy v.X.X required]

- **Design Pattern Categories**:
  - Aggregation queries (COUNT, SUM, AVG)
  - Cohort identification patterns
  - Longitudinal analysis patterns
  - Cross-sectional comparisons
  - [Verify patterns from actual query logs]

- **Healthcare Entity Categories**:
  - Clinical conditions (ICD-10 mappings)
  - Departments (organizational structure)
  - Data sources (EHR, claims, lab systems)
  - Provider specialties
  - [Internal Doc: Entity Mapping Specification]

#### Implementation Requirements
```python
# Example structure - verify actual implementation
classification_config = {
    "model": "[Specify model used]",
    "confidence_threshold": "[Measured threshold]",
    "fallback_behavior": "[Document fallback logic]",
    "performance_metrics": {
        "accuracy": "[REQUIRED FROM TESTING]",
        "latency": "[MEASURED IN MS]",
        "false_positive_rate": "[CALCULATED FROM TEST DATA]"
    }
}
```

#### Frequency Distribution Analysis
**Purpose**: Auto-identify field semantics from value patterns

**Methodology** [Verify with actual implementation]:
1. **Value Pattern Recognition**
   - Numeric ranges → Identify measurement types
   - Date patterns → Temporal field classification
   - Code patterns → Link to standard terminologies
   - [Document pattern matching rules]

2. **Statistical Profiling**
   ```sql
   -- Example approach - replace with actual queries
   SELECT 
     column_name,
     COUNT(DISTINCT value) as cardinality,
     MIN(value), MAX(value), AVG(value),
     STDDEV(value)
   FROM [table]
   GROUP BY column_name
   ```
   [Internal Doc: Profiling methodology required]

3. **Business Interpretation Mapping**
   - High cardinality + alphanumeric → Likely identifier
   - Low cardinality + specific values → Likely category
   - Date range patterns → Event timestamps
   - Numeric ranges → Measurements or counts
   - [Validation data required for each rule]

#### Catalog Generation Metrics [Require Measurement]
- [ ] Fields auto-classified correctly: __%
- [ ] Manual corrections required: __%
- [ ] Processing time per table: __ seconds
- [ ] Coverage of data types: __%

### What to Extract (Verify from Source)
- Classification model architecture
- Training data requirements
- Confidence scoring methodology
- Multi-class vs multi-label approaches
- Performance optimization techniques
- [All specifications must reference notebook sections]

### Required Test Data
1. [ ] Sample of 1000+ actual YuiQuery queries
2. [ ] Ground truth labels for classification accuracy
3. [ ] Data dictionary for catalog validation
4. [ ] Performance benchmarks from production

### Success Metrics [Must Measure]
- Query classification accuracy by category
- Catalog generation completeness
- Processing time per classification
- Resource utilization metrics
- Error rates and failure modes

## How to Reference in Whitepaper

**Acceptable:**
```
"Classification methodology adapted from Anthropic's guide 
[Source: https://github.com/anthropics/anthropic-cookbook/blob/main/skills/classification/guide.ipynb]
with healthcare-specific taxonomies [Internal Doc: Healthcare Taxonomy v.X.X]"

"Frequency distribution analysis identified field types with X% accuracy
[Internal Test: #123, Date: YYYY-MM-DD]"
```

**NOT Acceptable:**
- Claiming classification accuracy without test data
- Estimating catalog coverage without measurement
- Using "intelligent" or "smart" without specific metrics
- Projecting capabilities beyond tested scenarios

## Required Documentation Before Publishing

1. [ ] Classification taxonomy definitions (all categories)
2. [ ] Test results with confusion matrices
3. [ ] Data catalog generation accuracy report
4. [ ] Performance benchmarks under load
5. [ ] Field interpretation validation data
6. [ ] Query routing decision trees
7. [ ] Error handling procedures

## Compliance Checklist
- [ ] All classification categories documented
- [ ] Accuracy metrics from actual testing
- [ ] No estimated or projected performance
- [ ] Clear limitations documented
- [ ] Fallback behaviors specified
- [ ] Source citations for methodology

## Risk Mitigation
- **Misclassification Impact**: Document consequences and manual review triggers
- **Catalog Errors**: Validation workflow before production use
- **Performance Degradation**: Load testing results required
- **Privacy Concerns**: Ensure no PHI in training data

## Note
Classification capabilities must be demonstrated with measured performance on actual YuiQuery data. Theoretical capabilities or untested features should not be included in the whitepaper.

**Comment by stharrold** [comment]
Duplicate of #199
